### PR TITLE
[DOC-3124] Added details for secret expiry date

### PR DESCRIPTION
**Pull request by rashminandasahoo**
Set expiry date for secrets managed by Azure Key Vault.

Preview - https://6481bd098c65061db8d986fd--harness-developer.netlify.app/docs/platform/Secrets/add-use-text-secrets

## What Type of PR is This?

- [ ] Issue
- [x ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by nishantsaini20 — DISMISSED** [review]

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 1]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
+<img src={add_encrypted_text} alt="add_encrypted_text" height="200" width="500"/>
 ```
 
-Select the **Secret Manager** you will use to encrypt this secret.
+4. Select the **Secret Manager** you will use to encrypt this secret.
+
+5. In **Secret Name**, enter a name for the encrypted text. This is the name you will use to reference the text elsewhere in your resources.
 
-In **Secret Name**, enter a name for the encrypted text. This is the name you will use to reference the text elsewhere in your resources.
+6. In **Inline Secret** **Value**, enter a value for the encrypted text.
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:53]: Combine steps 6 and 7 - something like: 
6. Select the type of secret, and then configure the related values:
     * **Inline Secret Value**: enter a val.....
     * **Reference Secret**: if you have an Azure Key Vault or Hashicorp Vault secrets managers, you can reference existing secrets.

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 2]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
-In **Inline Secret** **Value**, enter a value for the encrypted text.
+   You can reference existing secrets in the following types of Secret Managers:
 
-#### Option: Reference Secret
+   * Azure Key Vault
+   * Hashicorp Vault
 
-You can create a Harness secret that refers to an existing secret by selecting **Reference** **Secret** and using that secret's name.
+8. A secret can have an expiry date if it is managed by Azure Key Vault. To set an expiry date, select a date in **Expires on**.
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:62]: Is this only if they select **Reference Secret**?

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 3]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
+   
+   :::important
+   The Harness Delegate version 79306 is required for this feature.
+   :::
 
-* Azure Key Vault
-* Hashicorp Vault
 
-Enter **Description** for your secret.
+9. Enter **Description** for your secret.
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:73]:
```suggestion
9. (Optional) Enter a **Description** and **Tags** for your secret.
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 4]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
-Enter **Tags** for your secret.
+10. Enter **Tags** for your secret.
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:75]:
```suggestion
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 5]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
-Click **Save.**
+11. Select **Save.**
 
-### Step 2: Use the Encrypted Text in Connectors
+## Use the Encrypted Text in Connectors
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:79]:
```suggestion
## Use the secret in connectors
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 6]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
 All of the passwords and keys used in Harness Connectors are stored as Encrypted Text secrets in Harness.
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:81]:
```suggestion
All of the passwords and keys used in Harness connectors are stored as encrypted text secrets in Harness.
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 7]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
 You can either create the Encrypted Text secret first and then select it in the Connector or you can create/select it from the Connector by clicking **Create or Select a Secret**:
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:83]:
```suggestion
You can either [create the secret](#add-a-text-secret) first and then select it in the connector. or you can create it from the connector by clicking **Create or Select a Secret**:
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 8]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
 ```mdx-code-block
-<img src={create_edit_encrypted_text} alt="create_edit_encrypted_text" height="300" width="700"/>
+<img src={create_edit_encrypted_text} alt="create_edit_encrypted_text" height="200" width="500"/>
 ```
 You can also edit it in the Connector.
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:88]:
```suggestion
You can also edit the secret in the connector.
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 9]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
 ```mdx-code-block
-<img src={edit_encrypted_text} alt="edit_encrypted_text" height="300" width="700"/>
+<img src={edit_encrypted_text} alt="edit_encrypted_text" height="200" width="500"/>
 ```
-### Step 3: Reference the Encrypted Text by Identifier
+## Reference the Encrypted Text by Identifier
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:93]:
```suggestion
## Reference the secret by identifier
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 10]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
 
 
 :::important
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:96]:
```suggestion
:::info important
```

Comment (StrawberryPad) [reply docs/platform/Secrets/2-add-use-text-secrets.md:96]: I think the above will make the note say "Important" but better test it.

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 11]
```diff
@@ -112,7 +124,7 @@ Avoid using `$` in your secret value. If your secret value includes `$`, you mus
 For example, if your secret in the Project scope has a value `'my$secret'`, and identifier `doc-secret`, to echo, use single quotes:  
 `echo '<+secrets.getValue("doc-secret")>'`
 
-### Review: Invalid Characters in Secret Names
+## Invalid Characters in Secret Names
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:127]:
```suggestion
## Invalid characters in secret names
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 12]
```diff
@@ -121,7 +133,7 @@ The following characters aren't allowed in the names of secrets:
  ~ ! @ # $ % ^ & * ' " ? / < > , ;
 ```
 
-### Review: Secrets in Outputs
+## Secrets in Outputs
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:136]:
```suggestion
## Secrets in outputs
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 13]
```diff
@@ -132,15 +144,15 @@ If you accidentally use a very common value in your secret, like whitespace, the
 
 If you see an output like this, review your secret and fix the error.
 
-### Review: Secret Scope
+## Secret Scope
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:147]:
```suggestion
## Secret scope
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 14]
```diff
@@ -132,15 +144,15 @@ If you accidentally use a very common value in your secret, like whitespace, the
...
 
 When creating secrets, it's important to understand their scope in your Harness account.
 
 A user can only create a secret according to the scope set by its Harness User permissions.
 
 For example, when you create a new project or a new organization, a Harness Secret Manager is automatically scoped to that level.
 
-### Review: Line breaks and Shell-Interpreted Characters
+## Line breaks and Shell-Interpreted Characters
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:155]:
```suggestion
## Line breaks and shell-interpreted characters
```

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 15]
```diff
@@ -163,10 +175,10 @@ MySecret:<+secrets.getValue("test")>
 EOF
 ```
 
-### Review: Characters length limit in Secret Names
+## Characters length limit in Secret Names
```
Comment (StrawberryPad) [inline docs/platform/Secrets/2-add-use-text-secrets.md:178]:
```suggestion
## Characters length limit in secret names
```

**Review by StrawberryPad — CHANGES_REQUESTED** [review]

File: docs/platform/Secrets/2-add-use-text-secrets.md [hunk 16]
```diff
@@ -13,72 +13,84 @@ import add_text_secret from './static/add-use-text-secrets-45.png'
...
+import secret_expiry_date from './static/azurekeyvault-select-date.png'
 ```
 
-You can add a text secret to the Secret Manager and use them in your resources like Pipelines and Connectors.
+You can add a text secret to the secret manager and use them in your resources like pipelines and connectors.
 
 This topic describes how to add a text secret in Harness.
 
-### Step 1: Add Text Secret
+## Add a text secret
```
[the remark this answers is not in the document]
Comment (rashminandasahoo) [reply docs/platform/Secrets/2-add-use-text-secrets.md:23]: As discussed, secrets can be added at any scope. Changing the heading would be misleading.

**Comment by rashminandasahoo** [comment]
Addressed the comments. Merging this since this is urgent.

**Pull request merged**